File: GuardianMonsters/guardians/src/de/limbusdev/guardianmonsters/guardians/abilities/IAbilityService.java
package de.limbusdev.guardianmonsters.guardians.abilities;

import de.limbusdev.guardianmonsters.guardians.Element;
import de.limbusdev.guardianmonsters.guardians.GuardiansServiceLocator;

/**
 * IAbilitiesService
 *
 * @author Georg Eckert 2017
 */

public interface IAbilityService extends GuardiansServiceLocator.Service
{
    Ability getAbility(Element e, int index);
    Ability getAbility(Ability.aID aID);
}

File: GuardianMonsters/guardians/src/de/limbusdev/guardianmonsters/guardians/abilities/IAbilityService.kt
package de.limbusdev.guardianmonsters.guardians.abilities

import de.limbusdev.guardianmonsters.guardians.Element
import de.limbusdev.guardianmonsters.guardians.GuardiansServiceLocator

/**
 * IAbilitiesService
 *
 * @author Georg Eckert 2019
 */

interface IAbilityService : GuardiansServiceLocator.Service
{
    fun getAbility(e: Element, index: Int): Ability
    fun getAbility(aID: Ability.aID): Ability
}
